modules/core/shared/src/main/scala/data/SemispaceCache.scala

```diff
@@ -12,31 +12,26 @@ import cats.syntax.all._
  */
 sealed abstract case class SemispaceCache[K, V](gen0: Map[K, V], gen1: Map[K, V], max: Int, possiblyEvicted: List[V]) {
 
-  assert(max >= 0)
```

Comment:
Requiring max >= 1 allows us to bypass the case where inserting results in immediate eviction. Allowing immediate eviction means the statement ends up getting closed before it can be used.